### DISCUSSION TITLE
Upgrade with different method than installation (broke longhorn)

### DISCUSSION THREAD

**Discussion by vinid223** [longhorn/longhorn#4556] [Q&A] [answered]
Hello there,

I would like to upgrade longhorn on my cluster, but I installed longhorn with helm using Rancher a long time ago.

The problem is that I do not use Rancher anymore because it slows down my cluster with the monitoring and is not reliable (the docker image sometimes just stops working) and everytime I tried to upgrade longhorn with Rancher/helm it just fails in the process. It breaks everything and then I need to spend hours trying to restart the pods that got detatched from the longhorn volumes or reboot nodes because longhorn have issues reconnecting with the storage medias.

I was wondering, can I change the upgrade/installation method to the kubectl with the manifest files ? I would hope to do so without needing to completely remove longhorn and manually redownload backups and set the volumes for each of my deployments.

I don't mind scaling down everything to 0 replicas for that process as long that I don't need to recreate my cluster in urgency because of that.

**Comment by joshimoo** [comment, marked as answer]
You can switch from the catalog install to the regular `kubectl apply -f *` deployment, I use this approach for my dev environment most of the time.

It's easiest if you do it as part of an upgrade, so that you don't run into any issues with the images since the catalog ones will use a different group / host.

In this case to limit the scope it's also best to just scale down the workloads and do the upgrade during a maintenance window.

**Reply by vinid223** [reply]
Thanks for the info, I tried but now longhorn can't start come csi-provisionner and my pods can't mount volumes. I get the following error 

`MountVolume.SetUp failed for volume "kube-api-access-rb76d" : [failed to fetch token: Post "https://127.0.0.1:6444/api/v1/namespaces/pihole/serviceaccounts/default/token": net/http: TLS handshake timeout, failed to sync configmap cache: timed out waiting for the condition]`

**Reply by vinid223** [reply]
I can see it tries to use the port 6444, but I think my cluster is accessed with the k3s port 6443

**Reply by vinid223** [reply]
I might be wrong here

**Comment by vinid223** [comment]
For more detail, the instance manager are unknown 
<img width="734" alt="image" src="https://user-images.githubusercontent.com/777832/190925698-f874ac72-0dcd-4c0b-a4ce-ab90c9c7e81d.png">


I did try to delete them and force them to go back up. It seems to work but I am not sure if it works.

Then there is theses 3 pods that keep crashing 
<img width="1099" alt="image" src="https://user-images.githubusercontent.com/777832/190925723-295c1468-fc13-4d5a-899c-4e7d875c49d6.png">


I have some longhorn manager working and csi-provisioner working.

**Comment by vinid223** [comment]
It took forever, but with patience, lots of restarts I finally managed to make the pods start. I guess I needed to be extremely patient and kill some stuck pods for longhorn
